akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentActorContext.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.scaladsl
+
+import akka.actor.typed.internal.adapter.ActorContextAdapter
+import akka.actor.typed.scaladsl.ActorContext
+import akka.persistence.typed.internal.Running
+
+object PersistentActorContext {
+
+  /**
+   * Access the to details about the persistent actor from the actor context, can only be used in response
+   * to a command, will throw `IllegalState` for other cases
+   */
+  def apply(context: ActorContext[_]): PersistentActorContext = {
+    context match {
+      case impl: ActorContextAdapter[_] ⇒
+        impl.currentBehavior match {
+          case w: Running.WithStateAccessible ⇒
+            new PersistentActorContext {
+              def lastSequenceNr: Long = w.state.seqNr
+            }
+
+          case s ⇒ throw new IllegalStateException(s"Cannot extract the PersistentActorContext in state ${s.getClass.getName}")
+
+        }
+
+      case c ⇒ throw new IllegalStateException(s"Cannot extract the PersistentActorContext from context ${c.getClass.getName}")
+    }
+  }
```

Comment:
it might be conceptually confusing that the ordinary ActorContext is only "retrieved" once via setup but the PersistentActorContext must be retrieved each time to hold latest information.

I was originally thinking that we would have something like `EventSourcedBehavior.setup(pcxt =>...)` and then we would update that pctx instance, but that might be difficult to implement?

Reply:
We could get away from that conceptual mismatch by not calling it `PersistentActorContext`, but have a static helper method: 
```
object EventSourcedBehavior {
  def currentSequenceNumber(context: ActorContext): Long
}
```

Reply:
Yes, that's a good idea.